src/petals/models/llama/llama_flash_attn_monkey_patch.py

```diff
@@ -0,0 +1,117 @@
+
+
+# Graciously stolen from the legend himself, John Smith.
+# https://raw.githubusercontent.com/johnsmith0031/alpaca_lora_4bit/winglian-setup_pip/src/alpaca_lora_4bit/monkeypatch/llama_flash_attn_monkey_patch.py
+
+
+from typing import List, Optional, Tuple
+import torch
+from torch import nn
+import transformers
+from einops import rearrange
+import math
+
+try:
+    from flash_attn.flash_attn_interface import flash_attn_func, flash_attn_varlen_qkvpacked_func
+    from flash_attn.bert_padding import unpad_input, pad_input
+except ImportError:
+    raise ImportError("Please install flash_attn to use this module")
+
+
+def replace_llama_attn_with_flash_attn():
+    transformers.models.llama.modeling_llama.LlamaAttention.forward = flash_attn_forward_gqa
+    print("Replaced attention with flash_attn")
```

Comment:
We do not use print statements inside the library code, please use logger.debug as in other modules

Reply:
Done, removed print and formatted with black and isort.